addon/mixins/data-adapter-mixin.js

```diff
@@ -56,6 +57,44 @@ export default Ember.Mixin.create({
   authorizer: null,
 
   /**
+    First authorizes the request and then calls base class' `ajax()`.
+
+    @param {String} url The requested URL.
+    @param {String} type The method of the HTTP request.
+    @param {Object} options Additional XHR options.
+  */
+  ajax(url, type, options) {
+    // tests benefit from doing it like this (instead of just `options = options || {}`).
+    if (!options) {
+      options = {};
+    }
+    const authorizer = this.get('authorizer');
+    Ember.assert("You're using the DataAdapterMixin without specifying an authorizer. Please add `authorizer: 'authorizer:application'` to your adapter.", Ember.isPresent(authorizer));
+
+    // Because of the way _super() works, we need to store references to base
+    // class implementation here in order to be able to use it in the callback below.
+    const me = this;
+    const supr = this._super;  // for ember 2.x
+    const nextsuper = this.__nextSuper;  // for ember 1.x
```

Comment:
I agree but after debugging this for couple of hours (both, in command line and in the browser) I found no other way. In Ember 1.x, `me` and `supr` references change somehow when the promise callback is executed. If you step in the debugger into the super function implementation it will never find the base class implementation (REST Adapters `ajax()` function) and will always just return `undefined`. I tried all kinds of tricks (using `function.bind(this)` for example).

Why this happens is beyond me. Maybe Babel can affect this too? Because those references shouldn't just change. It all works if I try calling `supr()` before the async part. 

I don't like this either but so far I've found no other way unfortunately. Using `__nextSuper` is clearly private and undocumented property to outside access but now we only use it in Ember 1.x so maybe we can get away with it?

Again, if anybody knows a better way of doing this, I'm more than happy to change this.